api/oc_core_res.c

```diff
@@ -589,6 +589,9 @@ core_get_resource_memory_by_index(int type, size_t device)
   if (type < OCF_CON) {
     return &g_core_resources[type];
   }
+  if (device >= OC_ATOMIC_LOAD32(g_device_count)) {
```

Comment:
add line to CoreGetResourceByIndex_F to test with invalid device index

Reply:
added